mlir-tensorrt/test/python/mlir_tensorrt_runtime/test_create_memref.py

```diff
@@ -514,3 +514,82 @@ def create_dangling_memref():
 # CHECK-LABEL: Test memref maintains data's lifetime
 # CHECK-NEXT: -- Inner scope: np.from_dlpack(): [1 2]
 # CHECK-NEXT: -- Outer scope: np.from_dlpack(): [1 2]
+
+
+def check_non_canonical_stride(client, assert_canonical_strides):
+    try:
+        t = cp.arange(12, dtype=cp.float32).reshape(3, 4)
+        a = cp.transpose(t)
+        memref = client.create_memref_view_from_dlpack(
+            a.__dlpack__(), assert_canonical_strides
+        )
+        if assert_canonical_strides:
+            print(
+                "Test failed: Expected exception was not thrown for CuPy non-canonical stride"
+            )
+        else:
+            print(
+                "Test passed: Correct error message received for non-canonical stride"
+            )
+        return False
+    except Exception as e:
+        error_message = str(e)
+        if (
+            "Given strides do not match canonical stride for the shape.\nShape: 4, 3\nExpected canonical stride: 3, 1\nGiven stride: 1, 4'"
+            in error_message
+        ):
+            print(
+                "Test passed: Correct error message received for non-canonical stride"
+            )
+            return True
+        else:
+            print(
+                f"Test failed: Unexpected error message for non-canonical stride: {error_message}"
+            )
+            return False
+
+
+def check_canonical_stride(client, assert_canonical_strides):
+    try:
+        t = cp.arange(12, dtype=cp.float32).reshape(3, 4)
+        memref = client.create_memref_view_from_dlpack(
+            t.__dlpack__(), assert_canonical_strides
+        )
+        print("Test passed: No exception thrown for canonical stride")
+        return True
+    except Exception as e:
+        print(f"Test failed: Unexpected exception for canonical stride: {str(e)}")
+        return False
+
+
+def test_memref_strides():
+    print("Testing non-canonical stride: assert_canonical_strides = True")
+    non_canonical_result = check_non_canonical_stride(
+        client, assert_canonical_strides=True
+    )
+
+    print("Testing non-canonical stride: assert_canonical_strides = False")
+    non_canonical_result = check_non_canonical_stride(
+        client, assert_canonical_strides=False
+    )
+
+    print("Testing canonical stride: assert_canonical_strides = True")
+    canonical_result = check_canonical_stride(client, assert_canonical_strides=True)
+
+    print("Testing canonical stride: assert_canonical_strides = False")
+    canonical_result = check_canonical_stride(client, assert_canonical_strides=False)
+
+
+print("Test memref strides")
+test_memref_strides()
+
+# CHECK-LABEL: Test memref strides
+# CHECK-NEXT: Testing non-canonical stride: assert_canonical_strides = True
```

Comment:
It looks like the error message is not being correctly matched. Maybe just print out the error message, e.g.

```# CHECK-NEXT: Testing non-canonical stride: assert_canonical_strides = True
# CHECK-NEXT: Received error message: .... specific message string ....```

No reason to do string matching in the Python exception, just do it in the `CHECK` statements?

Reply:
Agreed. I have updated to only use CHECK statements.